crates/core/src/host/wasm_common/module_host_actor.rs

```diff
@@ -787,9 +730,9 @@ impl<T: WasmInstance> WasmModuleInstance<T> {
                 let TableDef {
                     table_name,
                     columns,
-                    indexes: _,
                     table_type,
                     table_access,
+                    ..
```

Comment:
It was intentional to make the pattern match exhaustive, so that changes must be considered here. 

It does not seem like this comparison is correct anymore, because `columns` are not necessarily in the order of their `ColId`. We should probably collect both sides into sets before comparing.

It is also the case that the new fields in `TableDef` are not considered at all for the module update, namely sequences and constraints. If we want to support altering those when the row schema hasn't changed, this needs to be implemented. Otherwise, `eq` needs to return `false` if they're different, so as to reject the module update.

Reply:
See https://github.com/clockworklabs/SpacetimeDB/pull/267#discussion_r1355568416